src/ngraph/runtime/gpu/gpu_cuda_kernel_builder.cpp

```diff
@@ -1856,19 +1946,53 @@ void runtime::gpu::CudaKernelBuilder::coordinate_transform_to_multi_d(codegen::C
     //  product = product % stride[0]
     //  d1 = product/stride[1]
     //  ...
-    writer << "int coordinate_product = " << i_coord_product << ";\n";
+    writer << "int " << o_coordinates << "product = " << i_coord_product << ";\n";
     for (size_t i = 0; i < rank; i++)
     {
         if (i != 0)
         {
-            writer << "coordinate_product -= (" << o_coordinates << i - 1 << " * " << i_strides
-                   << brace_open << i - 1 << brace_close << ");\n";
+            writer << o_coordinates << "product -= (" << o_coordinates << i - 1 << " * "
+                   << i_strides << brace_open << i - 1 << brace_close << ");\n";
         }
         writer << "int " << o_coordinates << i << " = division_by_invariant_multiplication("
-               << "coordinate_product, " << i_stride_magic << brace_open << i << brace_close << ", "
-               << i_stride_shift << brace_open << i << brace_close << ");\n";
+               << o_coordinates << "product, " << i_stride_magic << brace_open << i << brace_close
+               << ", " << i_stride_shift << brace_open << i << brace_close << ");\n";
     }
 }
+
+void runtime::gpu::CudaKernelBuilder::collective_coordinate_transform_helper(
```

Comment:
Can we just modify the existing coordinate transform to support a `std::string reduced_idx` param? It can be a default parameter if you prefer not to change other uses of the coordinate transform.

Reply:
combined.